freshmaker/handlers/botas/botas_shipped_advisory.py

```diff
@@ -270,24 +306,107 @@ def handle(self, event):
                 to_rebuild_digests.add(digest)
 
         if not to_rebuild_digests:
-            msg = f"No bundle images to rebuild for advisory {event.advisory.name}"
+            msg = f"No bundle images to rebuild for advisory {self.event.advisory.name}"
             self.log_info(msg)
             db_event.transition(EventState.SKIPPED, msg)
             db.session.commit()
             return []
 
-        builds = self._prepare_builds(db_event, bundles_by_digest,
-                                      to_rebuild_digests)
+        bundles_to_rebuild = list(map(lambda x: bundles_by_digest[x],
+                                      to_rebuild_digests))
+        return bundles_to_rebuild
 
-        # Reset context to db_event.
-        self.set_context(db_event)
+    def _handle_manual_rebuild(self, db_event):
+        """
+        Handle manual rebuild submitted by Release Driver for an advisory created by Botas
 
-        self.start_to_build_images(builds)
-        msg = f"Advisory {db_event.search_key}: Rebuilding " \
-              f"{len(db_event.builds.all())} bundle images."
-        db_event.transition(EventState.BUILDING, msg)
+        :param db_event: database event that represent rebuild event
+        :rtype: list
+        :return: list of advisories that should be rebuilt
+        """
+        default_bundle_data = {
+            'nvr': '',
+            'update': {},
+            'append': {},
+            'pullspecs': [],
+        }
+        # map with old pullspec and bundle info
+        old_pullspec_to_bundle_map = dict()
```

Comment:
Optional:
```suggestion
        old_pullspec_to_bundle_map = {}
```

Reply:
I prefer using `dict()` because for me it's kind of more readable. When you what to know what is `old_pullspec_to_bundle_map`, it straight away says that it's a `dict`. But with `{}`, I know that sets can be created with that kind of parenthesis, and if I forgot that empty `{}` means `dict` It could take me some time to remember or I can just overlook it.

Reply:
No problem. `{}` is faster since the Python bytecode doesn't need to find the global of `dict`. I find it nicer to read but it's up to you. There's no style guide for Freshmaker so it's a free for all.  :smile: